lib/redux/actions/async.js

```diff
@@ -463,6 +470,43 @@ export function clickDeviceSelectionDone() {
   };
 }
 
+export function clickEditUserNext(profile) {
+  return (dispatch, getState) => {
+    const { uploadTargetUser } = getState();
+    const { api } = services;
+    dispatch(syncActions.updateProfileRequest());
+    if (!_.isEmpty(profile)){
+      api.user.updateProfile(uploadTargetUser, profile, (err, profile) => {
+        if (err) {
+          dispatch(syncActions.updateProfileFailure(err));
+        } else {
+          dispatch(syncActions.updateProfileSuccess());
+          async.series([
+            api.user.account,
+            api.user.loggedInProfile,
+            api.user.getUploadGroups
+          ], (err, results) => {
+            dispatch(syncActions.setAllUsers(results[0],results[1],results[2]));
+            const { targetDevices, devices, allUsers, loggedInUser } = getState();
+            const targetedDevices = _.get(targetDevices, uploadTargetUser, []);
+            const supportedDeviceKeys = Object.keys(devices);
+            const atLeastOneDeviceSupportedOnSystem = _.some(targetedDevices, (key) => {
+              return _.includes(supportedDeviceKeys, key);
+            });
+            if (_.isEmpty(targetedDevices) || !atLeastOneDeviceSupportedOnSystem) {
+              return dispatch(syncActions.setPage(pages.SETTINGS));
+            } else {
```

Comment:
Hmm, this is interesting. This action creator should only be available to clinics anyway, right? But say we later expose it to home users as well, then we may end up with unexpected behavior (not updating the "page"). I guess what I'm saying is, wouldn't it be fine to get rid of the `if` on line 499 and just always redirect to `pages.MAIN` in the `else`?